source/grid/manifold_lib.cc

```diff
@@ -462,6 +465,13 @@ get_new_point (const ArrayView<const Point<spacedim>> &vertices,
           }
       }
 
+    // If all the points are close to the candidate, we expect the candidate to
+    // be good enough. This tolerance was chosen such that the first iteration
+    // for a at least three time refined HyperShell mesh with radii .5 and 1.
+    // doesn't already succeed.
+    if (max_distance < 1.e-2)
+      return center + rho * candidate;
```

Comment:
This can't be the correct check -- `max_distance` has been computed in real space (not on the reference cell), i.e., it is sensitive to the size of the domain. Wouldn't this check, for example, always succeed if we are on a sphere of radius 1e-3?

Reply:
Both `candidate` (in line 406) and the `direction`s (in line 437) are normalized. This should make the check independent of the radius as far as I see.

Reply:
Oh, I missed this -- I had looked for something that pulls these points back to the reference cell. Thanks for pointing out my mistake. Would you mind adding a parenthetical remark to the comment that mentions that `max_distance` is the distance between points on the *unit* sphere, so an absolute tolerance is ok?